gapis/trace/android/trace.go

```diff
@@ -215,32 +235,65 @@ func (t *androidTracer) Validate(ctx context.Context) error {
 
 		startFunc(ctx)
 		if !doneSignal.Wait(ctx) {
-			err = log.Err(ctx, err, "Fail to wait for done signal from Perfetto.")
+			err = log.Err(ctx, err, "Failed to wait for done signal from Perfetto.")
+			timeout = true
 		}
 		log.I(ctx, "Perfetto trace size %v bytes", written)
 	})
+	if timeout == true {
```

Comment:
Can you just use `if err != nil` here and not need the additional boolean?

Reply:
Done